### DISCUSSION TITLE
Modify asset materialization - add additional metrics

### DISCUSSION THREAD

**Discussion by geoHeil** [dagster-io/dagster#19834] [Q&A]
How can I (from a delayed async operation) modify some existing asset materialization and update it? I.e. I have a cloud provider which is NOT providing accurate real time cost. Only some days later the real cost will be available.

How can I add the additional metrics/insights to dagster to correlate it nicely with the asset? I will need to update /add some key-value pairs to an existing materialization several days after the successful execution of a job

**Comment by maximearmstrong** [comment]
Hey @geoHeil - You could call `instance.report_runless_asset_event`, see documentation [here](https://docs.dagster.io/_apidocs/internals#dagster.DagsterInstance.report_runless_asset_event), that allows you to add an asset event that doesn't belong to a Dagster run.

**Reply by HPicatto** [reply]
In our case it is metadata specific to an asset run. so it's not related to the asset in general, and would not like it to be overridden

**Reply by geoHeil** [reply]
@maximearmstrong I do not want to report a new AssetMaterialization event just refine an existing one it looks like this method would only create a new one?

**Reply by geoHeil** [reply]
I just need to add some additional properties to the existing one
